ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageSievingInputStreamTest.java

```diff
@@ -51,37 +78,62 @@ public void resetMessageFormatHeaderVersionToUse() {
     MessageFormatRecord.headerVersionToUse = messageFormatHeaderVersionSaved;
   }
 
+  public MessageSievingInputStreamTest(EnumSet<TransformerOptions> options) throws Exception {
+    this.options = options;
+    this.storeKeyFactory = new MockIdFactory();
+    this.randomKeyConverter = new RandomKeyConverter();
+    if (options.contains(TransformerOptions.Validate)) {
+      transformers.add(new ValidatingTransformer(storeKeyFactory, randomKeyConverter));
+    }
+    if (options.contains(TransformerOptions.KeyConvert)) {
+      transformers.add(new ValidatingKeyConvertingTransformer(storeKeyFactory, randomKeyConverter));
```

Comment:
minor: is the `ValidatingKeyConvertingTransformer` required anymore or can you just use `BlobIdTransformer`?  (secondary concern but serves as a nice test for that transformer and its interaction with this class)

Reply:
I think @dharju or I can try doing that in a subsequent patch (things have to be moved around, as that class is in tools) - don't want to bloat this patch any further.

Reply:
If it is in tools, how will you inject it via configuration?

Reply:
i.e. just pointing out that it needs to be moved before it is usable

Reply:
It may have to be moved to the store - but injection is easy when `replication` or `server` is instantiating it (as long as the class methods are public, `MessageSievingInputStream` will be able to use it).

The `MessageSievingInputStreamTest` however cannot instantiate the `BlobIdTransformer` and there is no good place to put the latter in order for the former to access it.